finch-core/src/main/scala/io/finch/package.scala

```diff
@@ -147,11 +148,36 @@ package object finch {
    *   }
    * }}}
    *
-   * @param future a future to implicitly convert
+   * @param f a future to implicitly convert
    *
    * @return a service generated by ignoring the ''Req'' and returning the ''Future[Rep]''
    */
-  implicit def futureToService[Req, Rep](future: Future[Rep]): Service[Req, Rep] = new Service[Req, Rep] {
-    def apply(req: Req) = future
+  implicit def futureToService[Req, Rep](f: Future[Rep]): Service[Req, Rep] =
+    new Service[Req, Rep] {
+      def apply(req: Req) = f
+    }
+
+  /**
+   * A function that fetches the route information from the given request ''req''.
+   *
+   * @tparam Req the request type
+   */
+  trait Route[-Req] {
```

Comment:
We could use `@implicitNotFound` annotation with further explanation which why the implicit is required.